lib/daru/core/query.rb

```diff
@@ -39,7 +39,7 @@ def inspect
 
       class << self
         def apply_scalar_operator operator, data, other
-          BoolArray.new data.map { |d| d.respond_to?(operator) && !!d.send(operator, other) }
+          BoolArray.new data.map { |d| !!d.send(operator, other) if d.respond_to?(operator) }
```

Comment:
first checking whether operator works for `d` or not, if yes then only do `!!d.send(operator, other)`. (Actually for `nil` `d.respond_to?(operator)` is false when operator is any of these : [ '>' '<' , '=<' ,'>='], for `NaN` all these already works correctly)